Sources/Overlays/_Testing_WinSDK/Attachments/IWICBitmapSource+AttachableAsIWICBitmapSource.swift

```diff
@@ -0,0 +1,119 @@
+/// Because COM class inheritance is not visible in Swift, we must manually
+/// apply conformances to this protocol to each COM type that inherits from
+/// `IWICBitmapSource`.
+///
+/// Because this protocol is not `public`, we must also explicitly restate
+/// conformance to the public protocol `_AttachableByAddressAsIWICBitmapSource`
+/// even though this protocol refines that one. This protocol refines
+/// `_AttachableByAddressAsIWICBitmapSource` because otherwise the compiler will
+/// not allow us to declare `public` members in its extension that provides the
+/// implementation of `_AttachableByAddressAsIWICBitmapSource` below.
+///
+/// This protocol is not part of the public interface of the testing library. It
+/// allows us to reuse code across all subclasses of `IWICBitmapSource`.
+protocol IWICBitmapSourceProtocol: _AttachableByAddressAsIWICBitmapSource {}
+
+@_spi(Experimental)
+extension IWICBitmapSource: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+@_spi(Experimental)
+extension IWICBitmap: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+@_spi(Experimental)
+extension IWICBitmapClipper: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+@_spi(Experimental)
+extension IWICBitmapFlipRotator: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+@_spi(Experimental)
+extension IWICBitmapFrameDecode: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+@_spi(Experimental)
+extension IWICBitmapScaler: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+@_spi(Experimental)
+extension IWICBitmapSourceTransform2: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+@_spi(Experimental)
+extension IWICColorTransform: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+@_spi(Experimental)
+extension IWICFormatConverter: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+@_spi(Experimental)
+extension IWICPlanarFormatConverter: _AttachableByAddressAsIWICBitmapSource, IWICBitmapSourceProtocol {}
+
+// MARK: - Upcasting conveniences
+
+extension UnsafeMutablePointer where Pointee: IWICBitmapSourceProtocol {
+  /// Upcast this WIC bitmap to a WIC bitmap source (its parent type).
+  ///
+  /// - Returns: `self`, cast to the parent type via `QueryInterface()`. The
+  ///   caller is responsible for releasing the resulting object.
+  ///
+  /// - Throws: Any error that occurs while calling `QueryInterface()`. In
+  ///   practice, this function is not expected to throw an error as it should
+  ///   always be possible to cast a valid instance of `IWICBitmap` to
+  ///   `IWICBitmapSource`.
+  ///
+  /// - Important: This function consumes a reference to `self` even if the cast
+  ///   fails.
+  consuming func cast(to _: IWICBitmapSource.Type) throws -> UnsafeMutablePointer<IWICBitmapSource> {
+    try self.withMemoryRebound(to: IUnknown.self, capacity: 1) { `self` in
```

Comment:
The casts in this file to `IUnknown` are safe insofar as we use them to access fixed members of the COM vtable. The casts would become unsafe if we allowed the resulting pointers to escape _and_ if any of the types we use them on have multiple non-virtual inheritance to `IUnknown`.

Reply:
Can we capture this in a code comment for future reference?